projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/TransformJob.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { jobData } from '../jobData';
+import { VdkOption } from '../vdkOptions/vdk_options';
+import VDKTextInput from './VdkTextInput';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { jobTransformRequest } from '../serverRequests';
+import { IJobPathProp } from './props';
+import { VdkErrorMessage } from './VdkErrorMessage';
+
+export default class TransformJobDialog extends Component<IJobPathProp> {
+  /**
+   * Returns a React component for rendering a transform menu.
+   *
+   * @param props - component properties
+   * @returns React component
+   */
+  constructor(props: IJobPathProp) {
+    super(props);
+  }
+  /**
+   * Renders a dialog for transforming a data job.
+   *
+   * @returns React element
+   */
+  render(): React.ReactElement {
+    return (
+      <>
+        <VDKTextInput
+          option={VdkOption.NAME}
+          value="default-name"
+          label="Job Name:"
+        ></VDKTextInput>
+        <VDKTextInput
+          option={VdkOption.TEAM}
+          value="default-team"
+          label="Job Team:"
+        ></VDKTextInput>
+        <VDKTextInput
+          option={VdkOption.PATH}
+          value={this.props.jobPath}
+          label="Path to job directory:"
+        ></VDKTextInput>
+      </>
+    );
+  }
+}
+
+export async function showTransformJobDialog() {
+  const result = await showDialog({
+    title: 'Transform Job',
+    body: (
+      <TransformJobDialog
+        jobPath={jobData.get(VdkOption.PATH)!}
+      ></TransformJobDialog>
+    ),
+    buttons: [Dialog.okButton(), Dialog.cancelButton()]
+  });
+  if (result.button.accept) {
```

Comment:
Maybe adding a dialog asking "Are you sure you want to transform the job?" with cancel and yes buttons is a good idea here

Reply:
Why would we need such confirmation window when the transform operation would be working both ways (worst case the user would have to transform it back with a few clicks)? I am not sure this is necessary.

Reply:
For doing breaking changes like delete and update putting such dialogs are good.

Reply:
Alright. I suppose it would be better to have it then. Added it.